lib/_http_common.js

```diff
@@ -217,15 +217,20 @@ function checkIsHttpToken(val) {
   return tokenRegExp.test(val);
 }
 
-const headerCharRegex = /[^\t\x20-\x7e\x80-\xff]/;
+const headerCharRegex = /([^\t\x20-\x7e\x80-\xff])/;
```

Comment:
FWIW the capturing group makes things unnecessarily slower. You can get the same info from the return value of `exec()`: `[0]` contains the bad character and `.index` contains the index of the bad character.